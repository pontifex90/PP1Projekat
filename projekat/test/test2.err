ERROR 2017-02-11 06:18:03,733 [main][MJParser.java:778] : Syntax error "sd" on line 6, column 6
ERROR 2017-02-11 06:18:03,736 [main][MJParser.java:778] : Syntax error "da" on line 10, column 8
ERROR 2017-02-11 06:18:03,738 [main][MJParser.java:778] : Syntax error "44" on line 12, column 9
ERROR 2017-02-11 06:18:03,739 [main][MJParser.java:625] : All error messages follow:
				Syntax error "sd" on line 6, column 6
				Syntax error "da" on line 10, column 8
				Syntax error "44" on line 12, column 9
ERROR 2017-02-11 06:18:03,739 [main][MJParser.java:626] : Parsing finished with ERRORS!
