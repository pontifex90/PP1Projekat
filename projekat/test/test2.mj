
program test2

int b;

int a sd;
{

	void main(int a)
	int aa da;
	{
		b = 34 44 4;
	}
}

