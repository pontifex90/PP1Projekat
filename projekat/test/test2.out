INFO  2017-02-11 06:18:03,716 [main][MJParser.java:615] : Compiling source file: D:\PP1ProjekatRepo\PP1Projekat\projekat\test\test2.mj
INFO  2017-02-11 06:18:03,728 [main][MJParser.java:588] : #2 program
INFO  2017-02-11 06:18:03,729 [main][MJParser.java:588] : #49 test2
INFO  2017-02-11 06:18:03,729 [main][MJParser.java:588] : #49 int
DEBUG 2017-02-11 06:18:03,730 [main][MJParser.java:770] : Semantic info on line 2: added program "test2" to symbol table. [ kind: 6  name: test2  type: 0]
INFO  2017-02-11 06:18:03,732 [main][MJParser.java:588] : #49 b
INFO  2017-02-11 06:18:03,732 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:18:03,732 [main][MJParser.java:770] : Semantic info on line 4: added global variable "b" to symbol table. [ kind: 1  name: b  type: 1]
INFO  2017-02-11 06:18:03,733 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:18:03,733 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:18:03,733 [main][MJParser.java:588] : #49 sd
INFO  2017-02-11 06:18:03,733 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:18:03,733 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:18:03,733 [main][MJParser.java:588] : #17 void
DEBUG 2017-02-11 06:18:03,734 [main][MJParser.java:770] : Semantic info on line 6: added global variable "sd" to symbol table. [ kind: 1  name: sd  type: 1]
DEBUG 2017-02-11 06:18:03,735 [main][MJParser.java:874] : Error recovery successful: global variable declaration error (2)
INFO  2017-02-11 06:18:03,735 [main][MJParser.java:588] : #49 main
INFO  2017-02-11 06:18:03,735 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:18:03,735 [main][MJParser.java:770] : Semantic info on line 9: added  method "main" to symbol table. [ kind: 3  name: main  type: 0]
INFO  2017-02-11 06:18:03,735 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:18:03,736 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:18:03,736 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:18:03,736 [main][MJParser.java:770] : Semantic info on line 9: added formal parameter variable "a" to symbol table. [ kind: 1  name: a  type: 1]
INFO  2017-02-11 06:18:03,736 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:18:03,736 [main][MJParser.java:588] : #49 aa
INFO  2017-02-11 06:18:03,736 [main][MJParser.java:588] : #49 da
INFO  2017-02-11 06:18:03,736 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:18:03,736 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:18:03,736 [main][MJParser.java:588] : #49 b
DEBUG 2017-02-11 06:18:03,736 [main][MJParser.java:770] : Semantic info on line 10: added local variable "da" to symbol table. [ kind: 1  name: da  type: 1]
DEBUG 2017-02-11 06:18:03,736 [main][MJParser.java:874] : Error recovery successful: local variable declaration error (2)
INFO  2017-02-11 06:18:03,737 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:18:03,738 [main][MJParser.java:770] : Usage info on line 12: global variable "b" used. [ kind:1  name: b  type: 1]
INFO  2017-02-11 06:18:03,738 [main][MJParser.java:588] : #48 34
INFO  2017-02-11 06:18:03,738 [main][MJParser.java:588] : #48 44
INFO  2017-02-11 06:18:03,738 [main][MJParser.java:588] : #48 4
INFO  2017-02-11 06:18:03,738 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:18:03,738 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:18:03,738 [main][MJParser.java:588] : #8 }
DEBUG 2017-02-11 06:18:03,738 [main][MJParser.java:874] : Error recovery successful: Assign operator error (1)
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog test2: notype, 0, 1 
   Var b: int, 0, 0 
   Var sd: int, 1, 0 
   Meth main: notype, 0, 1 
      Var a: int, 0, 1 
      Var da: int, 1, 1 



INFO  2017-02-11 06:18:03,739 [main][MJParser.java:636] : 
===============================
VARS COUNTING:
INFO  2017-02-11 06:18:03,739 [main][MJParser.java:637] : Constants number: 0
INFO  2017-02-11 06:18:03,739 [main][MJParser.java:638] : Global vars number: 2
INFO  2017-02-11 06:18:03,739 [main][MJParser.java:639] : Global array vars number: 0
INFO  2017-02-11 06:18:03,739 [main][MJParser.java:640] : Class vars number: 0
INFO  2017-02-11 06:18:03,739 [main][MJParser.java:641] : Main function vars number: 1
===============================

INFO  2017-02-11 06:18:03,740 [main][MJParser.java:642] : 
===============================
FUNCTION CALLS COUNTING:
INFO  2017-02-11 06:18:03,740 [main][MJParser.java:643] : Global functions: 1
INFO  2017-02-11 06:18:03,740 [main][MJParser.java:644] : Class functions: 0
INFO  2017-02-11 06:18:03,740 [main][MJParser.java:645] : Class static functions: 0
INFO  2017-02-11 06:18:03,740 [main][MJParser.java:646] : Statements block number: 0
INFO  2017-02-11 06:18:03,740 [main][MJParser.java:647] : Main method function calls number: 0
INFO  2017-02-11 06:18:03,740 [main][MJParser.java:648] : Function formal parameters number: 1
INFO  2017-02-11 06:18:03,740 [main][MJParser.java:649] : Class definitions number: 0
